src/Orleans.Core.Abstractions/Core/Grain.cs

```diff
@@ -72,7 +78,7 @@ private static void ThrowInvalidContext()
         /// This constructor is particularly useful for unit testing where test code can create a Grain and replace
         /// the IGrainIdentity and IGrainRuntime with test doubles (mocks/stubs).
         /// </summary>
-        protected Grain(IGrainRuntime runtime) : this()
+        protected Grain(IGrainContext grainContext, IGrainRuntime runtime) : this()
```

Comment:
Shouldn't we use the `grainContext` passed in the constructor?

Reply:
Yes, you're right. I will flip the ctors around